File: Entidades/ClienteMayorista.vb
Public Class ClienteMayorista

    Public Property Id As Integer
    Public Property RazonSocial As String
    Public Property Nombre As String
    Public Property Cuit As String
    Public Property IdCondicionIva As Integer
    Public Property Comision As Integer?
    Public Property IdCorredor As Integer
    Public Property Transporte As String
    Public Property Bonificacion As Integer?
    Public Property Lista As Integer?
    Public Property CondicionPago As String
    Public Property IdEmpresa As Integer
    Public Property IdListaPrecio As Integer
    Public Property Observaciones As String
    Public Property IdDireccionFacturacion As Integer
    Public Property IdDireccionEntrega As Integer
    Public Property Habilitado As Boolean
    Public Property SaldoNotaPedido As Decimal
    Public Property SaldoCuentaCorriente As Decimal

    Sub New()
        Id = 0
        RazonSocial = String.Empty
        Nombre = String.Empty
        Cuit = Nothing
        IdCondicionIva = 0
        Comision = Nothing
        IdCorredor = 0
        Transporte = String.Empty
        Bonificacion = Nothing
        Lista = Nothing
        CondicionPago = String.Empty
        IdEmpresa = 0
        IdListaPrecio = 0
        Observaciones = String.Empty
        IdDireccionFacturacion = 0
        IdDireccionEntrega = 0
        Habilitado = False
        SaldoNotaPedido = 0
        SaldoCuentaCorriente = 0
    End Sub

End Class

File: Common/Common.Core/Model/ClienteMayorista.cs
using Common.Core.Enum;
using System;
using System.Reflection;

namespace Common.Core.Model
{
    public class ClienteMayorista : Entity<int>
    {
        public string RazonSocial { get; set; }
        public string Nombre { get; set; }
        public string Cuit { get; set; }
        public CondicionIVA CondicionIVA { get; set; }
        public decimal PorcentajeComision { get; set; }
        public decimal PorcentajeBonificacion { get; set; }
        public decimal PorcentajeLista { get; set; }
        public string CondicionPago { get; set; }
        public int IdListaPrecio { get; set; }
        public ListaPrecio ListaPrecio { get; set; }
        public int? IdDomicilioFacturacion { get; set; }
        public Domicilio DomicilioFacturacion { get; set; }
        public int? IdDomicilioEntrega { get; set; }
        public Domicilio DomicilioEntrega { get; set; }
        public string Observaciones { get; set; }
        public decimal MontoCuentaCorriente { get; set; }
        public decimal MontoNotasPedido { get; set; }
        public int? IdTransporte { get; set; }
        public Transporte Transporte { get; set; }
        public bool Habilitado { get; set; }

        public void DebitarSaldoCuentaCorriente(decimal monto)
        {
            MontoCuentaCorriente -= monto;
        }

        public void AcreditarSaldoCuentaCorriente(decimal monto)
        {
            MontoCuentaCorriente += monto;
        }
    }
}
